### PR TITLE
Log download failures

### PR DESCRIPTION
**Pull request by malena-ebert-sonarsource**

### CODE REVIEW

File: sonar-orchestrator/src/main/java/com/sonar/orchestrator/locator/ArtifactoryImpl.java [hunk 1]
```diff
@@ -98,8 +102,23 @@ public boolean downloadToFile(MavenLocation location, File toFile) {
         LOG.info("Found {} at {}", location, url);
         return Optional.of(toFile);
       } catch (HttpException e) {
-        if (e.getCode() != 404 && e.getCode() != 401 && e.getCode() != 403) {
+        if (e.getCode() != HTTP_NOT_FOUND && e.getCode() != HTTP_UNAUTHORIZED && e.getCode() != HTTP_FORBIDDEN) {
           throw new IllegalStateException("Failed to request " + url, e);
+        } else {
+          String errorMessage;
+          try {
+            JsonArray errors = Json.parse(e.getBody()).asObject().get("errors").asArray();
```
Comment (pierre-guillot-sonarsource) [inline sonar-orchestrator/src/main/java/com/sonar/orchestrator/locator/ArtifactoryImpl.java:110]: The error may to be json

Comment (malena-ebert-sonarsource) [reply sonar-orchestrator/src/main/java/com/sonar/orchestrator/locator/ArtifactoryImpl.java:110]: Can you clarify, what do you mean?

Comment (pierre-guillot-sonarsource) [reply sonar-orchestrator/src/main/java/com/sonar/orchestrator/locator/ArtifactoryImpl.java:110]: (sorry i missclicked the "resolve" button). Yeah so i mean that the payload return by the server may be non-json (like plain text). In that case, you will have a Json Parse Exception

Comment (pierre-guillot-sonarsource) [reply sonar-orchestrator/src/main/java/com/sonar/orchestrator/locator/ArtifactoryImpl.java:110]: aaaaand i realise that there is a catch for that :D so, nevermind, you did just fine :D

**Review by pierre-guillot-sonarsource — APPROVED** [review]
Looking good, you will need to add a UT to cover this. Let me know if you need some help to write it

**Review by pierre-guillot-sonarsource — APPROVED** [review]
Thanks for adding a UT :) That's a bit nitpicking, but maybe you could add as well one test with a non-json payload, to validate the catch of the parse exception.

**Comment by malena-ebert-sonarsource** [comment]
It is a bit tricked.

Both cases are handled.

Only one test case adds the JSON body. The others have none, and in this case there is a parsing error, which is handled by the catch block.

**Pull request merged**